packages/x-codemod/tsconfig.json

```diff
@@ -6,12 +6,11 @@
   },
   "include": [
     "src/**/*",
-    "../../node_modules/@mui/monorepo/packages/test-utils/src/initMatchers.ts",
+    "@mui-internal/test-utils/initMatchers.ts",
     "../../node_modules/@mui/material/themeCssVarsAugmentation",
-    "../../node_modules/date-fns",
-    "../../node_modules/dayjs/plugin/utc.d.ts",
-    "../../node_modules/dayjs/plugin/timezone.d.ts",
-    "../../node_modules/moment-timezone/index.d.ts"
+    "./node_modules/dayjs/plugin/utc.d.ts",
```

Comment:
Maybe out of scope for this PR, but feels wrong to `include` files that are outside of the source folder. Can't these be referenced using `compilerOptions.types`? Or from a declaration file that is included instead?

Reply:
@Janpot I've refactored all package tsconfig minimizing their content to the minimum and moved everything necessary to the `compilerOptions.types`.
Could you have another look? WDYT? 🤔